backend/app/Http/Controllers/RoomController.php

```diff
@@ -95,4 +93,23 @@ public function destroy(Request $request, Room $room): JsonResponse
         $room->delete();
         return response()->json(null, 204);
     }
+
+
+    /**
+     * Generate a random temporary room code for this room. 
+     */
+    public function generateRoomCode(Room $room): JsonResponse
+    {
+        if ($room->roomcode) {
+            return response()->json(['roomcode' => $room->roomcode]);
+        }
+
+        do {
+            $code = str_pad(random_int(0, 99999), 5, '0', STR_PAD_LEFT);
+        } while (Room::where('roomcode', $code)->exists());
```

Comment:
you need to add the checksum before checking it already exists. this check will alsways be false as there are no codes with length of five, only six